tests/Integration/Behaviour/Features/Context/Domain/CustomerGroupFeatureContext.php

```diff
@@ -0,0 +1,121 @@
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Exception;
+use PHPUnit\Framework\Assert;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Command\AddCustomerGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Query\GetCustomerGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryResult\EditableCustomerGroup;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+
+class CustomerGroupFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When /^I create a Customer Group "(.+)" with the following details:$/
+     *
+     * @param string $customerGroupReference
+     * @param TableNode $table
+     *
+     * @throws Exception
+     */
+    public function createCustomerUsingCommand(string $customerGroupReference, TableNode $table)
+    {
+        $data = $this->localizeByRows($table);
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddCustomerGroupCommand(
+            $data['name'],
+            new DecimalNumber($data['reduction']),
+            (bool) $data['displayPriceTaxExcluded'],
+            (bool) $data['showPrice'],
+            array_map(function (string $shopId) {
+                return new ShopId((int) $shopId);
+            }, explode(',', $data['shopIds']))
+        );
+
+        /** @var GroupId $id */
+        $id = $commandBus->handle($command);
+        $this->getSharedStorage()->set($customerGroupReference, $id->getValue());
+    }
+
+    /**
+     * @When /^I query Customer Group "(.+)" I should get a Customer Group with properties:$/
+     */
+    public function assertQueryCustomerProperties($customerGroupReference, EditableCustomerGroup $expectedGroup)
+    {
+        Assert::assertEquals($this->getCustomerGroupForEditing($customerGroupReference), $expectedGroup);
+    }
+
+    /**
+     * @Transform table:customer group,value
+     *
+     * @param TableNode $tableNode
+     *
+     * @return EditableCustomerGroup
+     */
+    public function transformEditableCustomerGroup(TableNode $tableNode): EditableCustomerGroup
+    {
+        $dataRows = $tableNode->getRowsHash();
+
+        return new EditableCustomerGroup(
+            (int) $dataRows['id'],
+            [
+                1 => $dataRows['name[en-US]'],
+                2 => $dataRows['name[fr-FR]'],
+            ],
```

Comment:
You should use `$data = $this->localizeByRows($table);`

Reply:
Done